tests/test_ip_address.py

```diff
@@ -148,3 +148,59 @@ def test_returns_failed_validation_on_invalid_ipv6_cidr_address(
 ):
     """Test returns failed validation on invalid ipv6 CIDR address."""
     assert isinstance(ipv6(address, cidr=cidr, strict=strict, host_bit=host_bit), ValidationError)
+
+
+@pytest.mark.parametrize(
+    ("address", "private"),
+    [
+        ("10.1.1.1", False),
+        ("192.168.1.1", False),
+        ("169.254.1.1", False),
+        ("127.0.0.1", False),
+        ("0.0.0.0", False),
+    ],
+)
+def test_returns_failed_validation_on_private_ipv4_address(address: str, private: bool):
+    """Test returns failed validation on invalid ipv4 CIDR address."""
+    assert isinstance(ipv4(address, private=private), ValidationError)
```

Comment:
Suggestion: I think the correct way is to use
```python
with pytest.raises(ValidationError):
    ipv4(adress,private=private)
```

Reply:
Ah, this must be then, done for every other test cases. You're welcome to make a PR.

Reply:
I can open a PR if you are interested

Reply:
Actually, it doesn't. It does not raise `ValidationError` by default. See: 

https://github.com/python-validators/validators/blob/d46cd586bdd6b9946d5d421f009794a25650f72e/src/validators/utils.py#L69-L70

So I think `pytest.raises` won't work.

Reply:
ah ok I see now.

If you set the env variable `RAISE_VALIDATION_ERROR` to `True` then a `ValidationError` is raised and `pytest.raises` will work. But if `RAISE_VALIDATION_ERROR=False`, then it is not going to work as the class `validationError` is returned.

Reply:
Right. Since the effect is the same before & after the change (except pytest action), I'll recommend to keep things as is.